src/mlpack/prereqs.hpp

```diff
@@ -73,6 +73,23 @@ using enable_if_t = typename enable_if<B, T>::type;
 #endif
 #endif
 
+// Backport std::any from C+17 to C++11 to replace boost::any.
+// Use mnmlstc backport implementation only if compiler does not
+// support C++17.
+#if __cplusplus <=201703L
```

Comment:
Should this be `<` instead of `<=`?  https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/ suggests that it will be set to `201703L` exactly when C++17 mode is used.